corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -327,6 +327,126 @@ hqDefine("geospatial/js/case_grouping_map",[
+        };
+
+        self.loadCaseGroups = function(caseGroups) {
+            self.allCaseGroups = caseGroups; // {case_id: {group_info}}
+            // Add groups to the cases being exported
+
+            var groupIds = [];
+            for (let caseID in caseGroups) {
+                let caseItem = caseGroups[caseID];
+
+                incrementGroupCaseCount(caseItem.groupId);
+                if (self.casePerGroup[caseItem.groupId] > max_cases_per_group) {
+                    self.groupMaxSizeBreached(true)
+                    return;
+                }
+                groupIds.push(caseItem.groupId);
+            }
+
+            let uniqueGroups = [];
+            new Set(groupIds).forEach(id => uniqueGroups.push(
+                {groupID: id, color: getRandomRGBColor()}
+            ));
+            self.allGroups(uniqueGroups);
+
+            let visibleIDs = _.map(uniqueGroups, function(group) {return group.groupID});
+            self.visibleGroupIDs(visibleIDs);
+            self.showAllGroups()
+        };
+
+        self.clear = function() {
+            self.allGroups([]);
+            self.visibleGroupIDs([]);
+        };
+
+        let incrementGroupCaseCount = function(groupId) {
+            if (!(groupId in self.casePerGroup)) {
+                self.casePerGroup[groupId] = 1;
+            } else {
+                self.casePerGroup[groupId] ++;
+            }
+        }
+
+        self.restoreMarkerOpacity = function() {
+            mapMarkers.forEach(function(marker) {
+                setMarkerOpacity(marker, DEFAULT_MARKER_OPACITY);
+            });
+        };
+
+        self.highlightGroup = function(group) {
+            exportModelInstance.casesToExport().forEach(caseItem => {
+                    let caseIsInGroup = caseItem.groupId === group.groupID;
+                    mapMarkers.forEach(function(marker) {
+                        let markerCoordinates = marker.getLngLat();
+                        let caseCoordinates = caseItem.coordinates;
+                        let latEqual = markerCoordinates.lat === caseCoordinates.lat;
+                        let lonEqual = markerCoordinates.lng === caseCoordinates.lng;
+                        let caseMarkerFound = latEqual && lonEqual;
+                        if (caseMarkerFound) {
+                            let opacity = DEFAULT_MARKER_OPACITY
+                            if (!caseIsInGroup) {
+                                opacity = 0.2;
+                            }
+                            setMarkerOpacity(marker, opacity);
+                        }
+                    });
+            });
+        };
+
+        function setMarkerOpacity(marker, opacity) {
+            let element = marker.getElement();
+            element.style.opacity = opacity;
+        };
+
+        self.showSelectedGroups = function() {
+            if (!self.allCaseGroups) {
```

Comment:
This will return truthy even if the object is empty `Boolean({}) === true`. It might be good to check for length here as well.

Reply:
JS fun:
![image](https://github.com/dimagi/commcare-hq/assets/64970009/dbfe2b57-b4fc-47f1-b9bc-f66c41a33c4e)
The `!{}` check is what we do here, so no need to check the length also

Reply:
Ah, JS always full of sneaky gotchas! All good then with this code :+1: